api/institutions/authentication.py

```diff
@@ -116,14 +131,29 @@ def authenticate(self, request):
             message = 'Institution SSO Error: invalid institution ID [{}]'.format(provider['id'])
             logger.error(message)
             sentry.log_message(message)
-            raise AuthenticationFailed(message)
+            raise PermissionDenied(detail='InstitutionSsoInvalidInstitution')
         username = provider['user'].get('username')
         fullname = provider['user'].get('fullname')
         given_name = provider['user'].get('givenName')
         family_name = provider['user'].get('familyName')
         middle_names = provider['user'].get('middleNames')
         suffix = provider['user'].get('suffix')
         department = provider['user'].get('department')
+        selective_sso_filter = provider['user'].get('selectiveSsoFilter')
+
+        # Check selective login first
+        is_selective_sso = provider['id'] in INSTITUTION_SELECTIVE_SSO_MAP
+        if is_selective_sso:
```

Comment:
Minor: If `is_selective_sso` isn't used anywhere else, this conditional could just be `if provider['id'] in INSTITUTION_SELECTIVE_SSO_MAP:`